vector/v.to.db/report.c

```diff
@@ -82,154 +152,350 @@ int report(void)
+            }
+            break;
+        case JSON:
+            for (i = 0; i < vstat.rcat; i++) {
+                record_value = json_value_init_object();
+                record = json_object(record_value);
+                json_object_set_number(record, "category", Values[i].cat);
+                json_object_set_number(record, "length", Values[i].d1);
+                json_array_append_value(records_array, record_value);
+                fsum += Values[i].d1;
+            }
+            json_object_dotset_number(root_object, "totals.length", fsum);
+            break;
+        }
+        break;
+    case O_SLOPE:
+        switch (format) {
+        case PLAIN:
             if (print_header)
-                fprintf(stdout, "cat%slength\n", options.fs);
+                fprintf(stdout, "cat%sslope\n", options.fs);
             for (i = 0; i < vstat.rcat; i++)
                 fprintf(stdout, "%d%s%.15g\n", Values[i].cat, options.fs,
                         Values[i].d1);
-        }
-        if (options.total) {
-            double sum = 0.0;
-
+            break;
+        case JSON:
             for (i = 0; i < vstat.rcat; i++) {
-                sum += Values[i].d1;
+                record_value = json_value_init_object();
+                record = json_object(record_value);
+                json_object_set_number(record, "category", Values[i].cat);
+                json_object_set_number(record, "slope", Values[i].d1);
+                json_array_append_value(records_array, record_value);
             }
-            fprintf(stdout, "total length%s%.15g\n", options.fs, sum);
+            break;
         }
-        break;
-    case O_SLOPE:
-        if (print_header)
-            fprintf(stdout, "cat%sslope\n", options.fs);
-        for (i = 0; i < vstat.rcat; i++)
-            fprintf(stdout, "%d%s%.15g\n", Values[i].cat, options.fs,
-                    Values[i].d1);
 
         break;
     case O_SINUOUS:
-        if (print_header)
-            fprintf(stdout, "cat%ssinuous\n", options.fs);
-        for (i = 0; i < vstat.rcat; i++)
-            fprintf(stdout, "%d%s%.15g\n", Values[i].cat, options.fs,
-                    Values[i].d1);
-        break;
+        switch (format) {
+        case PLAIN:
+            if (print_header)
+                fprintf(stdout, "cat%ssinuous\n", options.fs);
+            for (i = 0; i < vstat.rcat; i++)
+                fprintf(stdout, "%d%s%.15g\n", Values[i].cat, options.fs,
+                        Values[i].d1);
+            break;
+        case JSON:
+            for (i = 0; i < vstat.rcat; i++) {
+                record_value = json_value_init_object();
+                record = json_object(record_value);
+                json_object_set_number(record, "category", Values[i].cat);
+                json_object_set_number(record, "sinuous", Values[i].d1);
+                json_array_append_value(records_array, record_value);
+            }
+            break;
+        }
```

Comment:
```suggestion
        }
        break;
```

may perhaps fix the failed runner.